### PR TITLE
Added `const` where avaible

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
any perf change?

cc @WillAyd @jbrockmendel

**Review by jreback — APPROVED** [review]

File: pandas/_libs/tslibs/conversion.pyx [hunk 1]
```diff
@@ -595,8 +595,12 @@ cdef inline void localize_tso(_TSObject obj, tzinfo tz):
     obj.tzinfo = tz
 
 
-cdef inline bint _infer_tsobject_fold(_TSObject obj, ndarray[int64_t] trans,
-                                      int64_t[:] deltas, int32_t pos):
+cdef inline bint _infer_tsobject_fold(
+    _TSObject obj,
+    ndarray[int64_t] trans,
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/conversion.pyx]: can we make this `const int64_t[:]`?

**Comment by jbrockmendel** [comment]
LGTM

**Comment by MomIsBestFriend** [comment]
> any perf change?
> 
> cc @WillAyd @jbrockmendel

Unsure which function to test, which one you would like to see?

**Review by jreback — CHANGES_REQUESTED** [review]
asv on time zone functions

**Comment by MomIsBestFriend** [comment]
@jreback Actually I can't manage to run ```asv```, I hoped you can give me a specific function/method name to ```%timeit``` on

**Comment by MomIsBestFriend** [comment]
```
[ 75.00%] ··· tslibs.timestamp.TimestampOps.time_tz_convert                                                 ok
[ 75.00%] ··· ============ =============
                   tz                   
              ------------ -------------
                  None        296±4ns   
               US/Eastern    17.8±0.2μs 
                 <UTC>      5.11±0.04μs 
                tzutc()      5.17±0.2μs 
              ============ =============

[ 75.00%] · For pandas commit 4195bfbe <master> (round 2/2):
[ 75.00%] ·· Building for conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt...
[ 75.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[100.00%] ··· tslibs.timestamp.TimestampOps.time_tz_convert                                                 ok
[100.00%] ··· ============ =============
                   tz                   
              ------------ -------------
                  None        309±3ns   
               US/Eastern    17.6±0.4μs 
                 <UTC>      4.90±0.03μs 
                tzutc()      5.11±0.1μs 
              ============ =============


BENCHMARKS NOT SIGNIFICANTLY CHANGED.
```

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
Thanks @MomIsBestFriend

**Pull request merged**